Utils/Utils.hs

```diff
@@ -35,3 +35,12 @@ partition _ [x]      = [[x]]
 partition f (x:y:xs)
   | f x == f y = tack x (partition f (y : xs))
   | otherwise  = [x] : partition f (y : xs)
+
+-- Checks if two (unordered) lists contain the same elements
+-- E.g. listsEqual [3, 2, 1] [2, 1, 3] = True
+-- E.g. listsEqual [5, 5] [4, 4] = False
+listsEqual :: Eq a => [a] -> [a] -> Bool
+listsEqual []     []    = True
+listsEqual []     (_:_) = False
+listsEqual (x:xs) ys    = let ys' = delete x ys in
+  (length ys /= length ys') && listsEqual xs ys'
```

Comment:
I think it is a good idea to 1. rename this function, and 2. restructure it.

1. Rename it to something more or-pattern specific (e.g. `areOrBindingsEqual`); it is not expected to be used anywhere else and also we might want to refine this check later so it is better if its name is more specific. Also make its type more specific, no need for abstraction here.

2. About its implementation, instead of primitive recursion, I think it's a good idea to just split it into multiple separate, top-level checks (e.g. `(xs == nub xs) && (ys == nub ys) && (sort xs == sort ys)`). We might want to experiment a bit with how restrictive this function should be ;-)

Reply:
I tried using `sort`, but HsTmVar isn't an instance of `Ord` so that's why I decided to do this. I guess we can make it an instance of `Ord`, not too tricky I think.

Reply:
Yeah, feel free to add these instances
```
instance Ord (HsTyVar Sym)
instance Ord (HsTyVar Name)
```
I am surprised we didn't have them already!